lib/mix/test/mix/tasks/compile.protocols_test.exs

```diff
@@ -30,20 +31,22 @@ defmodule Mix.Tasks.Compile.ProtocolsTest do
       """)
 
       assert compile_elixir_and_protocols() == :ok
+      assert apply(Compile.Protocol, :__protocol__, [:impls]) == {:consolidated, [Integer]}
 
       assert mark_as_old!("_build/dev/lib/sample/consolidated/Elixir.Compile.Protocol.beam") !=
                @old
 
       # Delete a local implementation
       File.rm!("lib/impl.ex")
       assert compile_elixir_and_protocols() == :ok
+      assert apply(Compile.Protocol, :__protocol__, [:impls]) == {:consolidated, []}
 
       assert mark_as_old!("_build/dev/lib/sample/consolidated/Elixir.Compile.Protocol.beam") !=
                @old
 
       # Delete a local protocol
       File.rm!("lib/protocol.ex")
-      assert compile_elixir_and_protocols() == :noop
+      assert compile_elixir_and_protocols() == :ok
```

Comment:
`:ok` here and below is because `:protocols` now implies `:elixir` invocation which would return `{:ok, []}` not `{:noop, []}`. Checks for `Compile.Protocol.__protocol__(:impls)` now plays the role of real checker.